x/tx/signing/textual/any.go

```diff
@@ -113,3 +130,19 @@ func (ar anyValueRenderer) Parse(ctx context.Context, screens []Screen) (protore
 
 	return protoreflect.ValueOfMessage(anyMsg.ProtoReflect()), nil
 }
+
+// toAny converts the proto Message to a anypb.Any.
+// The input msg can be:
+// - either a anypb.Any already (in which case there's nothing to do),
+// - or a dynamicpb.Message.
+func toAny(msg proto.Message) (*anypb.Any, error) {
```

Comment:
In the case of a `dynamicpb.Messge`, this function and its brothers need to check that the dynamic message has an appropriate message name, not just one that has fields of the right name and type. (We could argue that the name was checked when we dispatched here, but it should be easy to check in this function.)

Even better, how about a single function `coerceToMessage(givenMsg, desiredMsg proto.Message) error` that initializes a given `desiredMsg` (presented as a protov2 concrete message), fast-paths the case where `givenMsg` is a protov2 concrete message of the same type, for a dynamicpb it checks that the names match then uses proto reflection to initialize the fields of `desiredMsg`, and otherwise throws an error. This should be much less error-prone and reduce the proliferation of `dynamicpb` everywhere.

Reply:
That's a good suggestion. I applied it in fbe2ce682c6912f659d1e654edd8285aaea7f2f3, and the code looks less scattered.

I'll see about moving `coerceToMessage` to cosmos-proto.